### DISCUSSION TITLE
Loading DSM (. Webp) terrain data using terrainlayer

### DISCUSSION THREAD

**Discussion by Paweled** [visgl/deck.gl#6080] [General]
Parameters I use:
                        meshMaxError:0,
			elevationDecoder:{
				rScaler: 65.536,
				gScaler: 0.256,
				bScaler: 0.001,
				offset: -6000
			},

However, a square grid will appear, and the effect is very unsatisfactory. Does anyone know how to solve it?
The following figure shows the grade 17 tile effect：
![image](https://user-images.githubusercontent.com/45220025/128834474-fceed4f7-9683-4cb6-8ac0-909c2c896150.png)

**Comment by kylebarron** [comment]
That's because you're using `meshMaxError` set to `0` I believe. That requires every triangle approximate the underlying grid _exactly_... which means the mesh ends up being exactly the underlying pixel-based terrain data.

**Reply by Paweled** [reply]
The following figure shows:wireframe:true
It should be connected by smooth triangles rather than stacked into cubes.
![image](https://user-images.githubusercontent.com/45220025/128967525-709b26bc-083b-4cb2-8472-854168a5f67a.png)

**Comment by Paweled** [comment]
I use polygonlayer to draw triangles as follows;
new PolygonLayer({
		data:pointData,
		pickable: false,
		filled: false,
		lineWidthMinPixels: 0,
		coordinateSystem: COORDINATE_SYSTEM.CARTESIAN,
		getPolygon: d => d.contour,
		getLineColor: [0,0,0],
		getFillColor: d =>[255,255,0,150],
		getLineWidth: 0.03
	})

![image](https://user-images.githubusercontent.com/45220025/128976244-6a1c7038-9d81-4dce-8db1-2b4c67371738.png)

**Comment by Paweled** [comment]
I think it should be the coordinate system.

I use WGS84 coordinates for boundaries in loaders.gl
I use polygonlayer to draw triangles as follows;

new PolygonLayer({
		data:pointData,
		pickable: false,
		filled: false,
		lineWidthMinPixels: 0,
		getPolygon: d => d.contour,
		getLineColor: [0,0,0],
		getFillColor: d =>[255,255,0,150],
		getLineWidth: 0.03
	})

![image](https://user-images.githubusercontent.com/45220025/128982142-b2d6be45-04db-4758-9b72-b9a99322b803.png)
